server/src/pages/homepage/controller.js

```diff
@@ -5,9 +5,27 @@ let model;
 
 exports.launch = function(m) {
   model = m;
+  if (window.wantsauth) {
+    if (window.wantsauth.getAuthData()) {
+      Object.assign(model, window.wantsauth.getAuthData());
+      updateModel();
+    } else {
+      window.wantsauth.addAuthDataListener((data) => {
+        Object.assign(model, data);
+        updateModel();
+        render();
+      });
+    }
+  }
   render();
 };
 
+function updateModel() {
+  if (!model.hasFxa) {
+    model.hasFxa = !!model.accountId;
+  }
+}
```

Comment:
nit: since this function might not be invoked at all, perhaps it's better as a function expression.